lib/Driver/Driver.cpp

```diff
@@ -1595,12 +1595,14 @@ void Driver::BuildInputs(const ToolChain &TC, DerivedArgList &Args,
           switch (FinalPhase) {
             // -E
             case phases::Preprocess:
-              if (Args.hasArg(options::OPT_cxxamp_kernel_mode)) {
-                Inputs.push_back(std::make_pair(types::TY_CXX_AMP, A));
-              } else if (Args.hasArg(options::OPT_cxxamp_cpu_mode)) {
+              if (Args.hasArg(options::OPT_cxxamp_cpu_mode))
                   Inputs.push_back(std::make_pair(types::TY_CXX_AMP_CPU, A));
+              if(Args.hasArg(options::OPT_hc_mode)) {
+                Inputs.push_back(std::make_pair(types::TY_HC_HOST, A));
+                Inputs.push_back(std::make_pair(types::TY_HC_KERNEL, A));
               } else {
```

Comment:
this `else` clause shall be better changed to test if we are in C++AMP mode. Otherwise "-E" would always have 2 inputs even if user doesn't specify `-std=c++amp`